_rules/SC1-3-1-aria-required-context-role.md

```diff
@@ -0,0 +1,198 @@
+---
+name: ARIA required context role
+description: | 
+  This rule checks that a role does not exist outside of its required context roles
+
+success_criterion:
+- 1.3.1 # Info and Relationships
+
+test_aspects: 
+- DOM Tree
+- CSS Styling
+
+authors:
+- Anne Thyme Nørregaard
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to any HTML or SVG element that is [included in the accessibility tree](#included-in-the-accessibility-tree) and has an explicit [semantic role](#semantic-role) with a [WAI-ARIA required context role](https://www.w3.org/TR/wai-aria-1.1/#scope), except if the element has an implicit semantic role that is identical to its explicit semantic role.
+
+### Expectation
+
+The [owner element](owner-element) for each target element has a [semantic role](#semantic-role) matching one of the [WAI-ARIA required context roles](https://www.w3.org/TR/wai-aria-1.1/#scope) for the target element.
+
+**Note:** The definition of [owner element](owner-element) used in this rule is diverging from the definition of ["owned element" in WAI-ARIA](https://www.w3.org/TR/wai-aria-1.1/#dfn-owned-element). See more in the [owner element](owner-element) definition.
+
+## Assumptions
+
+_There are currently no assumptions_
+
+## Accessibility Support
+
+This rule relies on assistive technologies to recognize [owner elements](#owner-element). This includes when the owner elements are ancestors that are not parents of the target element. However, some assistive technologies do not recognize owner elements that are not parents, unless workarounds are used.
+Furthermore, `aria-owns` has limited support in user agents.
+
+## Background
+
+- [Required Context Role](https://www.w3.org/TR/wai-aria-1.1/#scope)
+
+## Test Cases
+
+### Passed
+
+#### Passed example 1
+
+Element with role `listitem` is contained within its required context role `list`, expressed as an explicit role.
+
+```html
+<div role="list">
+    <div role="listitem"></div>
+</div>
+```
+
+#### Passed example 2
+
+Element with role `listitem` is contained within its required context role `list`, through the implicit role of `ul`.
+
+```html
+<ul>
+    <div role="listitem"></div>
+</ul>
+```
+
+#### Passed example 3
+
+Element contained within its required context role even though it is not a direct child of the context role.
+
+```html
+<div role="list">
+    <div>
+         <div>
+             <div>
+                <div role="listitem"></div>
+             </div>
+         <div>
+    <div>
+</div>
+```
+
+#### Passed example 4
+
+`aria-owns` used to give the target element the right context role.
+
+```html
+<div role="list" aria-owns="id1">
+  <div role="tabpanel">
+    <div id="id1" role="listitem"></div>
```

Comment:
Best to remove the tabpanel and move the listitem outside the list. That way you don't have an invalid tabpanel which may cause some tools to fail this.

Reply:
I have split the test case out in two: a simple one, like you recommended (Passed 4), and the more complex one with competing ownership.

Reply:
In the Example 5, I put text in the list item and I expected that VO reads "a list with 1 item". But it doesn't give us the semantic of the element.
```html
<div role="list" aria-owns="id1 id2">
    <div>Follow a list</div>
    <div id="id1" role="listitem">Lista 1</div>
    <div id="id2" role="listitem">Lista 2</div>
  </div>
```
With VO: how many items in the list? 3. I expected 2.

Either w/ a more natural building, VO doesn't recognise the aria-owns:
```html
<h2>Test 1: 1 native nested list w/ native relation</h2>

<ul>
<li>Fruts
  <ul>
    <li>Apples</li>
    <li>Bananas</li>
  </ul>
 </li>
 <li>Vegetables</li>
</ul>
  
  
<h2>Teste 2: 2 native lists w/ relations through aria-owns</h2>
  
<ul>
<li aria-owns="child">Fruts</li>
<li>Vegetables</li>
</ul>

<ul id="child">
<li>Apples</li>
<li>Bananas</li>
</ul>
  
<h2>Test 3: 2 non-native lists + relations through aria-owns and list semantic through ARIA</h2>

<div role="list">
<div role="listitem" aria-owns="fruit">Fruts</div>
<div role="listitem">Vegetables</div>
</div>

<div role="list" id="fruit">
<div role="listitem">Apples</div>
<div role="listitem">Bananas</div>
</div>
```
VO do not recognise the relations neither in test 2 nor test 3

Reply:
@jorgeponto, do you think this is covered well enough in the current Accessibility support section?

Reply:
@kasperisager and I discussed this, and we feel that it is covered through the following sections: 

> ## Accessibility Support
> 
>  - This rule relies on assistive technologies to recognize which elements are [owned by](#owned-by) each other. This includes when the element is owned by another element that is an ancestor, but not a parent of the target element. Some assistive technologies do not accept these owned by relationships, unless workarounds are used.
> Furthermore, `aria-owns` has limited support in some user agents.
> - Some user agents and assistive technologies ignore empty elements, which means they are not presented to all users. However, since this is handled inconsitently across user agents and assistive technologies, empty elements are applicable to this rule. 

"Owned by" definition: 

> **Note:** This definition is diverging from the definition of ["owned element" in WAI-ARIA](https://www.w3.org/TR/wai-aria-1.1/#dfn-owned-element). The reason is that the WAI-ARIA definition was found to provide too little guidance on how to handle specific edge cases where several elements compete about the ownership, and it seem that browser implementations of this are diverging a lot. This definition seeks to find a reasonable middle ground, but will have to be updated if the WAI-ARIA definition changes.

I am going to gamble, and go ahead and take this rule into Final Call, while waiting for @jorgeponto to get back on this.
If we find out, against expectations, that substantial changes are required to handle the cases of VO acting out, we will just have to redo the Final Call afterwards.